test/openssl/test_ssl.rb

```diff
@@ -2213,6 +2217,41 @@ def test_export_keying_material
     end
   end
 
+  def test_get_sigalg
+    # SSL_get0_signature_name() not supported
+    # SSL_get0_peer_signature_name() not supported
+    return unless openssl?(3, 5, 0)
+
+    server_proc = -> (ctx, ssl) {
+      assert_equal('rsa_pss_rsae_sha256', ssl.sigalg)
+      assert_nil(ssl.peer_sigalg)
+
+      readwrite_loop(ctx, ssl)
+    }
+    start_server(server_proc: server_proc) do |port|
+      cli_ctx = OpenSSL::SSL::SSLContext.new
+      server_connect(port, cli_ctx) do |ssl|
+        assert_nil(ssl.sigalg)
+        assert_equal('rsa_pss_rsae_sha256', ssl.peer_sigalg)
+        ssl.puts "abc"; ssl.gets
+      end
+    end
+  end
+
+  def test_group
+    # SSL_get0_group_name() not supported
+    return unless openssl?(3, 2, 0)
+
+    start_server do |port|
+      cli_ctx = OpenSSL::SSL::SSLContext.new
+      cli_ctx.groups = "X25519"
+      server_connect(port, cli_ctx) do |ssl|
+        assert_equal('x25519', ssl.group)
+        ssl.puts "abc"; ssl.gets
+      end
+    end
+  end
```

Comment:
I added assertions to test the `SSLContext#group` to the `test_set_groups_tls12` and `test_set_groups_tls12`. What do you think If you are okay with that, I will remove the test `test_group`.

Reply:
OK. I removed the `test_group`.